### ISSUE TITLE
Add privacy manifest

### ISSUE THREAD

**Issue by nikolay-fomichev-nl** [ably/ably-cocoa#1899]
Hello,

Last year during WWDC Apple announced privacy changes - https://developer.apple.com/videos/play/wwdc2023/10060.
In short: if an app OR an SDK uses certain iOS API then it must provide metadata file called privacy manifest. If an app or any of the SDK included in it fails to comply then Apple will reject app submission into App Store.

The deadline is May 1, 2024 - https://developer.apple.com/news/upcoming-requirements/?id=05012024a

It seems that Ably SDK uses at least one of the API in question, which is `NSUserDefaulta` - https://github.com/ably/ably-cocoa/blob/02d2a250a00f40eaa749fd6054d4e36f7c8490dc/Source/ARTLocalDeviceStorage.m#L21-L25

Could you please verify that this is the only one that is being used from the list - https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api and then provide privacy manifest file for the SDK and all of its dependencies if required.

### Which version of the Ably SDK are you using? 
- Ably - 1.2.21
- AblyDeltaCodec - 1.3.3
- msgpack - 0.4.0

### On which platform does the issue happen? 
All

### Are you using Carthage? 
n/a

### Are you using Cocoapods? 
1.12.1

### Which version of Xcode are you using? 
14.3 / 15.3

### What did you do?
Installed the SDK

### What did you expect to happen?
SDK should contain privacy manifest file

### What happened instead?
SDK does not contain privacy manifest file

Thank you

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4723) by [Unito](https://www.unito.io)

**Comment by ttypic** [comment]
Thank you very much for bringing this to our attention, @nikolay-fomichev-nl! We are looking into this

**Comment by umair-ably** [comment]
Released in [v1.2.30](https://github.com/ably/ably-cocoa/releases/tag/1.2.30)

**Comment by nikolay-fomichev-nl** [comment]
Hello. 
I updated `Ably` pod and noticed that privacy manifest is not downloaded while using Cocoapods (v1.15.2).
The file is in the repo - https://github.com/ably/ably-cocoa/blob/main/Source/PrivacyInfo.xcprivacy.
But it is not declared in podspec file - https://github.com/CocoaPods/Specs/blob/master/Specs/1/4/1/Ably/1.2.30/Ably.podspec.json

Here's an [example](https://github.com/Alamofire/Alamofire/blob/master/Alamofire.podspec#L28) of adding privacy manifest into the podpec file:
```
s.resource_bundles = {'Alamofire' => ['Source/PrivacyInfo.xcprivacy']}
```

**Comment by umair-ably** [comment]
Thanks for raising this @nikolay-fomichev-nl - I'll expedite how quickly we look at this given the deadline is today

**Comment by umair-ably** [comment]
Released in [v1.2.31](https://github.com/ably/ably-cocoa/releases/tag/1.2.31)

cc: @nikolay-fomichev-nl - thanks again!
